sdk/core/src/main/java/com/klaviyo/core/config/Config.kt

```diff
@@ -20,7 +20,9 @@ interface Config {
         fun applicationContext(context: Context): Builder
         fun debounceInterval(debounceInterval: Int): Builder
         fun networkTimeout(networkTimeout: Int): Builder
-        fun networkFlushInterval(networkFlushInterval: Int): Builder
+        fun networkFlushIntervalWifi(networkFlushInterval: Int): Builder
+        fun networkFlushIntervalCell(networkFlushInterval: Int): Builder
+        fun networkFlushIntervalOffline(networkFlushInterval: Int): Builder
```

Comment:
FYI we _should_ be able to just turn off the background thread altogether when offline, and restart it when we get a network-restored event. This would be a good fall back though if a device doesn't properly report connection change events... that is an area I need to do a little real-world testing on. In theory the `NewtorkMonitor` class is providing that functionality, but since we are supporting back to API 23 we should thoroughly test older APIs.

Reply:
Ya I had a similar thought about the offline state but just decided to follow what was written in the ticket as a fallback for the same reasons you are outlying here